src/parsing/response.rs

```diff
@@ -85,6 +86,7 @@ pub fn parse_response<B>(reader: BaseStream, request: &PreparedRequest<B>) -> Re
     headers.remove(TRANSFER_ENCODING);
 
     Ok(Response {
+        url: url.clone(),
```

Comment:
wouldn't `request.url()` be the same and avoid adding a param?

Reply:
nevermind, no, if there's a redirect it will be the wrong url

Reply:
Yes, I'll just need to update `self.url` in `send`'s loop.

Reply:
I think that what you are currently doing is better. I like to think of the PreparedRequest as immutable.